index.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+var RtmClient = require('@slack/client').RtmClient;
+var RTM_EVENTS = require('@slack/client').RTM_EVENTS;
+var CLIENT_EVENTS = require('@slack/client').CLIENT_EVENTS;
+var schedule = require('node-schedule');
+
+var { util } = require('./app');
+
+
+const botToken = process.env.SLACK_BOT_TOKEN || '';
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+
+if (!botToken || !darkskyToken) {
+    console.error('SLACK_BOT_TOKEN and DARKSKY_TOKEN must be passed as environment variables!');
+    process.exit(1);
+}
+
+var rtm = new RtmClient(botToken);
+
+let botID;
+let botChannel;
+
+rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, (rtmStartData) => {
+    // XXX the private surry-channel doesn't show up in the rtmStartData even though my
+    // bot was invited to the channel :(
+
+    // looks like self.id should always be present - https://api.slack.com/methods/rtm.start
+    botID = rtmStartData.self.id.toLowerCase();
+    console.log(`Logged in as ${rtmStartData.self.name}(${botID}) of team ${rtmStartData.team.name}, but not yet connected to a channel`);
+});
+
+let rtmConnectionOpened = false;
+
+rtm.on(RTM_EVENTS.MESSAGE, function handleRtmMessage(message) {
+
+    if (!message.text) {
+        return;
+    }
+
+    // forgive trailing spaces and ignore case
+    const normalizedMessage = message.text.trim().toLowerCase();
+    let now = false;
+    let tomorrow = false;
+
+    // only respond to messages where we are mentioned
+    if (normalizedMessage.indexOf(`<@${botID}>`) === -1) {
+        return;
+    }
+
+    // private channel ID isn't available on start, so respond on the channel
+    // we're mentioned on.
+    botChannel = message.channel;
+
+    if (normalizedMessage.indexOf('weather now') !== -1) {
+        now = true;
+    }
+
+    if (normalizedMessage.indexOf('weather tomorrow') !== -1) {
+        tomorrow = true;
+    }
+
+    if (!now && !tomorrow) {
+        if (rtmConnectionOpened) {
+            if (normalizedMessage.indexOf('help') !== -1) {
+                rtm.sendMessage(`<@${message.user}>, try @surry-interview 'weather now' or 'weather tomorrow'`, botChannel);
+            } else {
+                rtm.sendMessage(`<@${message.user}>, I did not understand that command, try @surry-interview 'weather now' or 'weather tomorrow'`, botChannel);
+                console.log('Ignoring unknown command: ' + normalizedMessage.substring(0, 20) + '...');
+            }
+        }
+        return;
+    }
+
+    if (now) {
+        util.getCurrentWeatherReport().then((json) => {
+            if (rtmConnectionOpened) {
```

Comment:
I'm not sure if chat messages can be received before the client state is 'connected', but the initial message following authentication with the initial user metadata is received before the client is fully connected: https://slackapi.github.io/node-slack-sdk/bots#posting-a-message

I went with the approach from my other comment where I moved the `rtmConnectionOpened` check into a separate function. I also realized the `rtm` client object has its own `connected` property, so I switched to that.